### PR TITLE
Add mocks for common node libraries 

### PR DESCRIPTION
**Pull request by SupremeTechnopriest**
add mock support for:
- `fs`
- `dns`
- `dgram`
- `child_process`
- `cluster`
- `module`
- `net`
- `readline`
- `repl`
- `tls`

add optional shim for:
- fs

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,2 @@
 dist
+node_modules
```
Comment (calvinmetcalf) [inline .gitignore]: you should just add node_modules to your global .gitignore

File: src/index.js [hunk 2]
```diff
@@ -1,9 +1,9 @@
-import {join} from 'path';
+import { join } from 'path';
 const libs = new Map();
 
 // our es6 versions
-libs.set('process', require.resolve('process-es6'));
```
Comment (calvinmetcalf) [inline src/index.js]: don't move this please

File: src/index.js [hunk 3]
```diff
@@ -1,9 +1,9 @@
-import {join} from 'path';
+import { join } from 'path';
```
Comment (calvinmetcalf) [inline src/index.js]: please avoid unneccisary whitespace changes

File: src/index.js [hunk 4]
```diff
@@ -30,16 +30,32 @@ libs.set('zlib', require.resolve(join('..', 'src', 'es6', 'zlib')));
...
+libs.set('tls', EMPTY_PATH);
+
+
 export default function (opts) {
   opts = opts || {};
   let cryptoPath = EMPTY_PATH;
+  let fsPath = EMPTY_PATH;
   if (opts.crypto) {
     cryptoPath = CRYPTO_PATH;
   }
```
Comment (calvinmetcalf) [inline src/index.js]: we should probably make crypto a shim otherwise

**Comment by SupremeTechnopriest** [comment]
@calvinmetcalf I'll make these changes.

File: .gitignore [hunk 5]
```diff
@@ -1,2 +1 @@
-dist
```
Comment (calvinmetcalf) [inline .gitignore]: you deleted the wrong line here

Comment (SupremeTechnopriest) [inline .gitignore]: sorry man, just working thought this.  Ill get it all cleaned up and let you know.  `npm link` does terrible things, so I have to push to github and `npm i` from my fork.

File: src/es6/readable-stream/readable.js [hunk 6]
```diff
@@ -7,7 +7,7 @@ import {inherits, debuglog} from 'util';
 import BufferList from './buffer-list';
 import {StringDecoder} from 'string_decoder';
 import {Duplex} from './duplex';
-import {nextTick} from 'process';
```
Comment (SupremeTechnopriest) [inline src/es6/readable-stream/readable.js]: also here ^^

File: src/es6/util.js [hunk 7]
```diff
@@ -18,7 +18,7 @@
 // DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
 // OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE
 // USE OR OTHER DEALINGS IN THE SOFTWARE.
-import process from 'process';
```
Comment (SupremeTechnopriest) [inline src/es6/util.js]: and here ^^

**Comment by SupremeTechnopriest** [comment]
@calvinmetcalf If you can sign off on those changes to the process requires I will tidy up.  I have everything working 100% on my complex project with these changes.

**Comment by SupremeTechnopriest** [comment]
Hey @calvinmetcalf I created another branch with the `.gitignore` changes you requested.  Personally I don't use global `.gitignore` or globally installed packages.  It is my belief that everything required to contribute to a project should be encompassed inside that project.  Just `npm i` and go.  Im going to close this PR and open a new one from my other branch.  I need the `dist` folder committed on my master branch since I'm installing from git until such time that these features are merged in.

**Pull request closed without merging**